cpp/src/arrow/ipc/reader.cc

```diff
@@ -954,6 +955,36 @@ Result<std::shared_ptr<SparseIndex>> ReadSparseCSXIndex(
   }
 }
 
+Result<std::shared_ptr<SparseIndex>> ReadSparseCSFIndex(
+    const flatbuf::SparseTensor* sparse_tensor, const std::vector<int64_t>& shape,
+    io::RandomAccessFile* file) {
+  auto* sparse_index = sparse_tensor->sparseIndex_as_SparseTensorIndexCSF();
+  const auto ndim = static_cast<int64_t>(shape.size());
+  auto* indptr_buffers = sparse_index->indptrBuffers();
+  auto* indices_buffers = sparse_index->indicesBuffers();
+  std::vector<std::shared_ptr<Buffer>> indptr_data(ndim - 1);
+  std::vector<std::shared_ptr<Buffer>> indices_data(ndim);
+
+  std::shared_ptr<DataType> indptr_type, indices_type;
+  std::vector<int64_t> indices_size, axis_order;
+
+  RETURN_NOT_OK(internal::GetSparseCSFIndexMetadata(
+      sparse_index, &indices_size, &axis_order, &indptr_type, &indices_type));
+
+  for (int64_t i = 0; i < ndim - 1; ++i) {
+    ARROW_ASSIGN_OR_RAISE(indptr_data[i], file->ReadAt(indptr_buffers->Get(i)->offset(),
+                                                       indptr_buffers->Get(i)->length()));
+  }
+  for (int64_t i = ndim - 1; i < 2 * ndim - 1; ++i) {
```

Comment:
`i` should start from `0` for `indices_data` and `indices_buffers`.

Reply:
Changed.
It seems the offset and length values here are either not correctly read from metadata or something goes wrong at write-time. Either way I'm getting:
```
'_error_or_value26.status()' failed with IOError: Read out of bounds (offset = 94719482752432, size = 1688849860263936) in file of size 264

```